src/idl_gen_rust.cpp

```diff
@@ -960,12 +960,20 @@ class RustGenerator : public BaseGenerator {
         return "bool";
       }
       case ftStruct: {
-        const auto typname = WrapInNameSpace(*type.struct_def);
-        return "Option<&" + lifetime + " " + typname + ">";
+        auto typname = WrapInNameSpace(*type.struct_def);
```

Comment:
Upon further thought, what do you think about making a C++ utility function with this signature?

```c++
std::string WrapInOptionIfNotRequired(std::string, bool)
```

That encapsulates this logic? that way we can abstract out all the `if (!field.required)` stuff and also simplify the patch.

Your code changes would then look like:

```c++
return WrapInOptionIfNotRequired("&" + lifetime + " " + typname, field.required);
```

Reply:
Yes, please refactor. Same with all similar looking code below.